windowsforms/FlexGridShowcaseDemo/cs/MainForm.cs

```diff
@@ -0,0 +1,689 @@
+﻿using C1.Win.FlexGrid;
+using C1.Win.Themes;
+using C1.Win.Ribbon;
+using C1.Win.RulesManager;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Windows.Forms;
+
+namespace FlexGridShowcaseDemo
+{
+
+    public partial class MainForm : C1RibbonForm 
+    {
+        private const string _customThemeName = "Greenwich";
+        private const int _footerTextPadding = 4;
+        private DataSet _dataSet = new DataSet();
+        private C1RulesManager _rulesManager = new C1RulesManager();
+        private IEnumerable<IRule> _rules = Enumerable.Empty<IRule>();
+        private static Random _rnd = new Random();
+
+        #region Initialization
+
+        public MainForm()
+        {
+            InitializeComponent();
+        }
+
+        private void MainForm_Load(object sender, EventArgs e)
```

Comment:
@RussKie That raises a question :smile:
When can having this event handler be useful if I can always just override OnLoad?

Reply:
You can't override unless you subclass.

Reply:
@RussKie yes but I think every form should already subclass the `System.Windows.Forms.Form` class. I can't think of a situation where I'll have SomeForm_Load event handler outside a class that inherits `System.Windows.Forms.Form`. Am I missing something?

Reply:
No, that's correct. If you feel lazy - you use the designer to wire up an event for you. Otherwise you write the method yourself.

The designer is a tool which makes the design process easier. However like any tool it shines in some areas, and underperforms in others. Like a hammer - you can use it to drive a nail in to a wall, but equally you could use it to drive a screw in... Though the latter is best done with a screwdriver.

Reply:
Thanks!
Probably worth a remark in the API doc?

https://github.com/dotnet/winforms/blob/2ae45c66fdcb8e6aadf43288b7429f17f1fd0f49/src/System.Windows.Forms/src/System/Windows/Forms/Form.cs#L2470-L2479

Reply:
Fixed with d6bec26f2e8d94674969099d15d0e436725d0e01.